### DISCUSSION TITLE
Apply Inverse Transform after Inferencing

### DISCUSSION THREAD

**Discussion by magnusziegler19** [TorchIO-project/torchio#478] [Q&A] [answered]
Hey Everyone,

I'm working on a project related to quality control with PC-MRI, and as part of that, I'm using torchio.
My question is related to how best to apply inverse transforms after inferencing data. Inferencing is currently being done for 3D patches derived from a larger 3D volume (input). The input volume however has been preprocessed to fit a standard size using the CropOrPad transform. I saw the function apply_inverse_transform() in the subject data structure but I am uncertain how to apply it here as the output is a tensor not a 'subject'.

Here is a snippet from my code for context:

```python
curr_subnum = inference_dataset[0]['subjectNum']
    with torch.no_grad():
        for image_num, image in enumerate(tqdm(inference_dataset)):
            # metadata
            image = inference_dataset[image_num]
            subnum = image['subjectNum']
            vel = image['velocity']
            imagenum = image['imageNum']
            output_image_filename = args.output_data_path + f'{subnum}_{vel}_unw_{imagenum:03}.nii'

            grid_sampler = tio.inference.GridSampler(sample=inference_dataset[image_num],
                                                     patch_size=args.patch_size,
                                                     patch_overlap=args.patch_overlap)
            patch_loader = torch.utils.data.DataLoader(grid_sampler, batch_size=args.patch_batch_size)
            aggregator = tio.inference.GridAggregator(grid_sampler)

            for patches_batch in patch_loader:
                input_tensor = patches_batch['MRA'][tio.DATA]
                locations = patches_batch[tio.LOCATION]
                output = model(input_tensor)  # Run batch through the network
                aggregator.add_batch(output, locations)

            output_image = aggregator.get_output_tensor()
           ## something to apply inverse transform here ## 
            output_image = sitk.GetImageFromArray(output_image.numpy())

            # write image to disk w/ the same metadata as it came in with
            sitk_writer.SetFileName(output_image_filename)
            sitk_writer.Execute(output_image)
```

So I'm a bit confused as to how best apply the inverse transform and would appreciate input.

Cheers,

Magnus

**Comment by fepegar** [comment, marked as answer]
Hi, @magnusziegler19. Apologies for the late reply, I just saw this! I agree this part of the library is still new and a bit confusing. Could you please take a look at the [inference tutorial](https://github.com/fepegar/torchio/tree/master/examples#inference) and let me know if you still have questions?
